ui/components/Modal.js

```diff
@@ -290,3 +291,110 @@ function Modal(props) {
 }
 
 export default React.memo(Modal);
+
+function RJSFModalWrapper({
+  handleClose,
+  schema,
+  uiSchema = {},
+  initialData = {},
+  handleSubmit,
+  title,
+  submitBtnText,
+  helpText,
+}) {
+  const formRef = useRef();
+  const classes = useStyles();
+  const formStateRef = useRef();
+  const [canNotSubmit, setCanNotSubmit] = useState(false);
+  const [snackbar, setSnackbar] = useState(false);
+  const [loadingSchema, setLoadingSchema] = useState(true);
+
+  useEffect(() => {
+    setCanNotSubmit(false);
+    const handleDesignNameCheck = () => {
+      const designName = title?.toLowerCase();
+      const forbiddenWords = ['untitled design', 'Untitled', 'lfx'];
+
+      for (const word of forbiddenWords) {
+        if (designName?.includes(word)) {
+          setSnackbar({
+            severity: 'warning',
+            message: `Design name should not contain Untitled Design, Untitled, LFX`,
+            open: true,
+          });
+          setCanNotSubmit(true);
+          break;
+        }
+      }
+    };
+    handleDesignNameCheck();
+  }, [title]);
+
+  const handleFormChange = (data) => {
+    formStateRef.current = data;
+  };
+
+  useEffect(() => {
+    setLoadingSchema(!schema);
+  }, [schema]);
+
+  const handleFormSubmit = () => {
+    if (formRef.current && formRef.current.validateForm()) {
+      handleSubmit(formRef.current.state.formData);
+      handleClose();
+    }
+  };
+
+  return (
+    <>
+      <ModalBody>
+        {loadingSchema ? (
+          <div style={{ textAlign: 'center', padding: '8rem 17rem' }}>
+            <CircularProgress />
+          </div>
+        ) : (
+          <RJSFWrapper
```

Comment:
@aabidsofi19 @Yashsharma1911  here what i am saying that just wrapping the existing rjsfwrapper with sistent modal

Reply:
extract the rjsf part only .  eg how will you handle the case for secondary and tertiary buttons ?

Reply:
we were till now using only i guess primary and secondary actions in these rjsf modals that is either we confirm or cancel

Reply:
@sudhanshutech  was this feedback acted upon ?

Reply:
for rjsf yes the rjsf wrapper is extracted and for the buttons slight modification needed so i have a upcoming pr shortly.